SimPEG/Directives.py

```diff
@@ -337,17 +338,24 @@ def endIter(self):
 
             self.mode = 2
 
+            mmap = self.reg.mapping * self.invProb.model
+
             # Either use the supplied epsilon, or fix base on distribution of
             # model values
-            if getattr(self, 'eps', None) is None:
-                self.reg.eps_p = np.percentile(np.abs(self.invProb.model), self.prctile)
-            else:
-                self.reg.eps_p = self.eps[0]
+            for imodel in range(self.reg.nModels):
```

Comment:
what is the `nModels`? should it be a property of the regularization? or does it make more sense to attach it to the directive? (eg. `self.nModels`)